codestodians/food-basket-api
# [Feat: #154163232] update key project dependencies

#### What does this PR do?
- It updates important project modules like `Django` and `graphene`.
- Fixes breaking changes as a result of the modules update.
- Update the models to ensure `null` values can be allowed in certain instances.

#### Description of Task to be completed?
- Update modules
- Fix breaking changes
- Update tests
- Update models
- Update schema

#### How should this be manually tested?
- Install updated modules with `pip3 install -r requirements.txt`
- Run the recent migration.
- Run queries and mutations on `graphiql` to test that everything is in order.
- Run tests with `coverage run --source=food_basket --omit */snapshots/* manage.py test`
- Check the coverage report with `coverage report -m`

#### Any background context you want to provide?
- The versions of Django and graphene have been updated a couple of times over the past months, and the new versions have better ways of solving certain problems. Updating this now should help in staying up to date with improvements in the usage of those modules/libraries.
- Also, the addition of new constraints to the models ensures that all values do not always have to be supplied in all cases.

#### What are the relevant pivotal tracker stories?
- [#154163232](https://www.pivotaltracker.com/story/show/154163232)
